ext/tags.go

```diff
@@ -106,11 +108,11 @@ type rpcServerOption struct {
 	clientContext opentracing.SpanContext
 }
 
-func (r rpcServerOption) Apply(o *opentracing.StartSpanOptions) {
+func (r rpcServerOption) Apply(t opentracing.Tracer, o *opentracing.StartSpanOptions) {
```

Comment:
I'm curious what you think of the user-visible API change, since Apply is not user-visible. If you like the user-visible change, do you see another way to do this?

I considered passing the Tracer directly to the CarriedXXXOption, which also feels not clean, because that makes it possible for the Extract tracer to differ from the StartSpan tracer (and requires the user to pass the same thing twice).

I basically agree with you, but I think a little uncleanliness in an internal API is better than requiring users to call Extract directly when they don't have any special error handling requirements. This will make the common case of StartSpan from a carried context a one-liner, which I think is important (and I generally don't think purely-diagnostic APIs should require users to handle errors).